machine/state.go

```diff
@@ -25,7 +25,7 @@ const (
 type MachineState struct {
 	ID       string
 	PublicIP string
-	Metadata map[string]string
+	Metadata map[string]string `json:"-"`
```

Comment:
We really should not let registry-level serialization logic leak out into the machine package. Please copy the MachineState struct over to where this JSON tag matters. We've already taken this approach with `unitModel` and `unit.UnitFile` in registry/unit.go.